pkgs/by-name/sm/smartcat/package.nix

```
@@ -0,0 +1,41 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+
+, openssl
+, pkg-config
+, libX11
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "smartcat";
+  version = "1.3.0";
+
+  src = fetchFromGitHub {
+    owner = "efugier";
+    repo = "smartcat";
+    rev = "refs/tags/${version}";
+    hash = "sha256-iCtNNKXo0peGGUaQXKaaYaEo7MAL70PX0BAWPERNmlo=";
+  };
+
+  cargoHash = "sha256-qNl2bI5VqpMfoFc+gZC4XHrNT9pnWseduYSOi5Dzr9M=";
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    openssl
+    libX11
+  ];
```

```suggestion
  ]
  ++ lib.optionals stdenv.isDarwin [
    darwin.apple_sdk.frameworks.AppKit
    darwin.apple_sdk.frameworks.Security
    darwin.apple_sdk.frameworks.SystemConfiguration
  ];
```

(dependencies from `nix-init`, so these might be superfluous or insufficient)

---

That's my bad, I didn't realize darwin support required manual intervention. I can't verify the binary itself works myself as I lack a machine to test with, but cross compiling worked at least.